components/datetime/src/provider/mod.rs

```diff
@@ -110,6 +110,20 @@ const _: () = {
     impl_datetime_patterns_date_roc_v1!(Baked);
 };
 
+#[cfg(feature = "compiled_data")]
+impl icu_provider::DataProvider<icu_time::provider::TimezonePeriodsV1> for Baked {
```

Comment:
I've concluded that this is fine and different from the other things handled by ExternalLoader, because you are loading the data struct, not a whole formatter. We need ExternalLoader because the return type is something like `DecimalFormatter` and we want to call `DecimalFormatter::try_new`.